File: python/raindrops/raindrops.py
def get_factors(number: int) -> list[int]:
    return [n for n in range(1, number + 1) if number % n == 0]


def convert(number: int) -> str:
    factors = get_factors(number)
    speak = ""
    if 3 not in factors and 5 not in factors and 7 not in factors:
        return str(number)
    if 3 in factors:
        speak += "Pling"
    if 5 in factors:
        speak += "Plang"
    if 7 in factors:
        speak += "Plong"
    return speak

File: go/raindrops/raindrops.go
package raindrops

import "fmt"

func contains(factors []int, number int) bool {
	for _, n := range factors {
		if n == number {
			return true
		}
	}
	return false
}

func getFactors(number int) []int {
	var factors []int
	for _, i := range []int{3, 5, 7} {
		if number%i == 0 {
			factors = append(factors, i)
		}
	}

	return factors
}

func Convert(number int) string {
	var sound string
	factors := getFactors(number)

	if !contains(factors, 3) && !contains(factors, 5) && !contains(factors, 7) {
		return fmt.Sprintf("%d", number)
	}

	if contains(factors, 3) {
		sound += "Pling"
	}
	if contains(factors, 5) {
		sound += "Plang"
	}
	if contains(factors, 7) {
		sound += "Plong"
	}

	return sound
}
